test/integration/consul-container/test/upgrade/peering_http_test.go

```diff
@@ -64,27 +132,33 @@ func TestPeering_UpgradeToTarget_fromLatest(t *testing.T) {
 		//  - Register a new static-client service in dialing cluster and
 		//  - set upstream to static-server service in peered cluster
 
-		// Restart the gateway & proxy sidecar
+		// Restart the gateway & proxy sidecar, and verify existing connection
 		require.NoError(t, dialing.Gateway.Restart())
-		require.NoError(t, dialing.Container.Restart())
-
 		// Restarted gateway should not have any measurement on data plane traffic
 		libassert.AssertEnvoyMetricAtMost(t, gatewayAdminPort,
 			"cluster.static-server.default.default.accepting-to-dialer.external",
 			"upstream_cx_total", 0)
+		libassert.HTTPServiceEchoes(t, "localhost", staticClientPort, "")
+
+		require.NoError(t, dialing.Container.Restart())
+		libassert.HTTPServiceEchoes(t, "localhost", staticClientPort, "")
+		// require.NoError(t, accepting.Container.Restart())
+		// libassert.HTTPServiceEchoes(t, "localhost", staticClientPort, "")
 
 		clientSidecarService, err := libservice.CreateAndRegisterStaticClientSidecar(dialingCluster.Servers()[0], libtopology.DialingPeerName, true)
 		require.NoError(t, err)
 		_, port := clientSidecarService.GetAddr()
 		_, adminPort := clientSidecarService.GetAdminAddr()
-		require.NoError(t, clientSidecarService.Restart())
 		libassert.AssertUpstreamEndpointStatus(t, adminPort, fmt.Sprintf("static-server.default.%s.external", libtopology.DialingPeerName), "HEALTHY", 1)
 		libassert.HTTPServiceEchoes(t, "localhost", port, "")
 		libassert.AssertFortioName(t, fmt.Sprintf("http://localhost:%d", port), "static-server")
+
+		// TODO: restart static-server-2's sidecar
+		tc.extraAssertion(appPort)
```

Comment:
what is extraAssertion doing here?

Reply:
In this case, the extra assertion is to assert the request is routed to a second static server, which is named as `static-server-2`.

Reply:
makes sense. Perhaps we need to rename this function to describe what is being asserted.